### ISSUE TITLE
BUG: LAG WITH TEXT IN ARROWS

### ISSUE THREAD

**Issue by HoodyEz** [zsviczian/obsidian-excalidraw-plugin#1893]
**Your environment**
SYSTEM INFO:
	Obsidian version: v1.6.5
	Installer version: v1.5.12
	Operating system: Windows 10 Home 10.0.22631
	Login status: not logged in
	Insider build toggle: off
	Live preview: off
	Base theme: light
	Community theme: none
	Snippets enabled: 11
	Restricted mode: off
	Plugins installed: 16
	Plugins enabled: 12
		1: Tray v0.3.5
		2: Custom File Explorer sorting v2.1.12
		3: Calendar v1.5.10
		4: Latex Suite v1.9.4
		5: Iconize v2.14.2
		6: Excalidraw v2.2.9
		7: Dataview v0.5.67
		8: Fleeting Notes Sync v0.6.3
		9: Opener v2.1.2
		10: Projects v1.17.3
		11: Spaced Repetition v1.12.4
		12: Tasks v7.6.1

**Describe the bug**
Basically, there is a lot of lag when I have arrows with text on the canvas. It's not just with the free draw tool but with all the elements in general; everything becomes very laggy.

**To Reproduce**
1. Place an arrow
2. Introduce text in the arrow
3. Feel the lag

**Screenshots**

https://github.com/user-attachments/assets/fd575bda-6f61-453a-aa7a-6fccc64cd9c9

**Additional context**
It happens with only 1 arrow with text, and it's worse with more arrows.

**Comment by zsviczian** [comment]
very interesting - This needs to be fixed in the Excalidraw package, not in the Obsidian plugin. I raised an issue for this: https://github.com/excalidraw/excalidraw/issues/8257

**Comment by zsviczian** [comment]
since another issue is now open for this, I will close this one. Let's hope someone picks this up and finds a solution.

**Comment by zsviczian** [comment]
I am deploying two fixes for this with 2.2.10 🤞
